apps/extension/src/content-scripts/message.ts

```diff
@@ -0,0 +1,33 @@
+import { Prax } from '../message/prax';
+
+// @ts-expect-error - ts can't understand the injected string
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type PraxMessage<T = unknown> = { [PRAX]: T };
```

Comment:
What if we just do:

```typescript
export type PraxMessage<T> = Record<typeof PRAX, T>;
const isPraxMessageEventData = <T>(p: unknown): p is PraxMessage<T> =>
  typeof p === 'object' && p != null && PRAX in p;
```

I don't think `{ [PRAX]: T }` is giving us typesafety anyway.

Reply:
it's useful to have a composeable type